x-pack/elastic-agent/pkg/agent/cmd/container.go

```diff
@@ -532,26 +532,26 @@ func runLegacyAPMServer(streams *cli.IOStreams, path string) (*process.Info, err
 	if release.Snapshot() {
 		version = fmt.Sprintf("%s-SNAPSHOT", version)
 	}
-	// Extract the bundled apm-server binary into the APM_SERVER_PATH
+	// Extract the bundled apm-server into the APM_SERVER_PATH
 	if err := installer.Install(context.Background(), spec, version, path); err != nil {
 		return nil, errors.New(err,
 			fmt.Sprintf("installing %s (%s) from %s to %s", spec.Name, version, cfg.TargetDirectory, path))
 	}
-
-	// Start apm-server process respecting args
-	logInfo(streams, "Starting legacy apm-server daemon as a subprocess.")
-	pattern := filepath.Join(path, fmt.Sprintf("%s-%s-%s*", spec.Cmd, version, cfg.OS()), spec.Cmd)
-	files, err := filepath.Glob(pattern)
+	// Get the apm-server directory
+	files, err := ioutil.ReadDir(path)
 	if err != nil {
-		return nil, errors.New(err, fmt.Sprintf("searching apm-server in %s", pattern))
+		return nil, errors.New(err, fmt.Sprintf("reading directory %s", path))
 	}
-	if len(files) != 1 {
-		return nil, errors.New("multiple apm-server versions installed")
+	if len(files) != 1 || !files[0].IsDir() {
+		return nil, errors.New("expected one directory")
 	}
-	f, err := filepath.Abs(files[0])
-	if err != nil {
-		return nil, errors.New(err, fmt.Sprintf("absPath for %s", files[0]))
+	apmDir := filepath.Join(path, files[0].Name())
+	// Extract the ingest pipeline definition to the HOME_DIR
+	if home := os.Getenv("HOME_PATH"); home != "" {
+		syncDir(filepath.Join(apmDir, "ingest"), filepath.Join(home, "ingest"))
```

Comment:
You don't catch the error here? I think that should be handled so if syncing the directory fails its clear.